float64[] xyz
---
bool can_reach